proposals/interface-types/working-notes/record.md

```diff
@@ -0,0 +1,73 @@
+
+
+### Records
+
+Records are collections of named and typed fields. Records can be defined in as
+Interface Types; for example this definition encodes a familiar notion of a
+credit card:
+
+```wat
+(@interface datatype @cc 
+  (record "cc"
+    (ccNo u64)
+    (name string)
+    (expires (record
+      (mon u8)
+      (year u16)
+      )
+    )
+    (ccv u16)
+  )
+)
+```
+
+Lifting and lowering a record involves lifting/lowering the individual fields of
+the record and then 'packing' or 'unpacking' the record as an ordered
+sub-sequence of elements on the stack.
+
+For example, the sequence:
+
+```wat
+...
+local.get $ccNo
+i64-to-u64
+local.get $name
+local.get $name.len
+memory-to-string
+i8.local.get $mon
+i8_to_u8
```

Comment:
IIUC, `$mon` is a wasm-core-typed local and thus this sequence would be:
```wasm
local.get $mon
i32-to-u8
```
It'd be nice to put this instruction sequence into a proper adapter function (with param/result) so that the types of these `$` variables is clear.